XPlat/Shopify/iOS/source/Shopify.iOS/ApiDefinition.cs

```diff
@@ -0,0 +1,1964 @@
+
+		[Export ("controller:failedToCompleteCheckout:withError:")]
+		void ControllerFailedToCompleteCheckout (ViewController controller, Checkout checkout, NSError error);
+
+		[Export ("controller:didCompleteCheckout:status:")]
+		void ControllerDidCompleteCheckout (ViewController controller, Checkout checkout, Status status);
+
+		[Export ("controller:didDismissApplePayControllerWithStatus:forCheckout:")]
+		void ControllerDidDismissApplePayController (ViewController controller, PKPaymentAuthorizationStatus status, Checkout checkout);
+
+		[Export ("controller:didDismissWebCheckout:")]
+		void ControllerDidDismissWebCheckout (ViewController controller, Checkout checkout);
+
+		[Export ("didDismissViewController:")]
+		void DidDismissViewController (ViewController controller);
+
+		[Export ("controllerWillCheckoutViaWeb:")]
+		void ControllerWillCheckoutViaWeb (ViewController viewController);
+
+		[Export ("controllerWillCheckoutViaApplePay:")]
+		void ControllerWillCheckoutViaApplePay (ViewController viewController);
+
+	}
+
+	[BaseType (typeof(UIViewController), Name = "BUYViewController")]
+	[DisableDefaultCtor]
+	interface ViewController : IPKPaymentAuthorizationViewControllerDelegate
+	{
+		[NullAllowed, Export ("delegate", ArgumentSemantic.Weak)]
+		IViewControllerDelegate Delegate { get; set; }
+
+		[Export ("client", ArgumentSemantic.Strong)]
+		BuyClient Client { get; set; }
+
+		[Export ("shop", ArgumentSemantic.Strong)]
+		Shop Shop { get; set; }
+
+		[Export ("merchantId", ArgumentSemantic.Strong)]
+		string MerchantId { get; set; }
+
+		[Export ("isApplePayAvailable")]
+		bool IsApplePayAvailable { get; }
+
+		[Export ("allowApplePaySetup")]
+		bool AllowApplePaySetup { get; set; }
+
+		[Export ("canShowApplePaySetup")]
+		bool CanShowApplePaySetup { get; }
+
+		[Export ("shouldShowApplePayButton")]
+		bool ShouldShowApplePayButton { get; }
+
+		[Export ("shouldShowApplePaySetup")]
+		bool ShouldShowApplePaySetup { get; }
+
+		[Export ("checkout", ArgumentSemantic.Strong)]
+		Checkout Checkout { get; }
+
+		[Export ("loadShopWithCallback:"), Async]
+		void LoadShop (Action<bool, NSError> block);
+
+		[Export ("supportedNetworks", ArgumentSemantic.Copy)]
+		NSString[] SupportedNetworks { get; set; }
+
+		[Export ("paymentRequest")]
+		PKPaymentRequest PaymentRequest { get; }
+
+		[Export ("initWithClient:")]
+		IntPtr Constructor (BuyClient client);
+
+		[Export ("startApplePayCheckout:")]
+		void StartApplePayCheckout (Checkout checkout);
+
+		[Export ("startWebCheckout:")]
+		void StartWebCheckout (Checkout checkout);
+
+		[Export ("startCheckoutWithCartToken:")]
+		void StartCheckout (string token);
+
+		[Export ("checkoutCompleted:status:")]
+		void CheckoutCompleted (Checkout checkout, Status status);
+
+		[Static]
+		[Export ("completeCheckoutFromLaunchURL:")]
+		void CompleteCheckout (NSUrl url);
+
+	}
+
+	[BaseType (typeof(ViewController), Name = "BUYProductViewController")]
+	interface ProductViewController : Themeable
```

Comment:
I would suggest to add [DisableDefaultCtor] attribute because it would make it easier to configure the theme using Constructor (BuyClient client, Theme theme).